src/app/debug/debug-tree/debug-tree.component.ts

```diff
@@ -129,94 +119,74 @@ export class DebugTreeComponent implements AfterViewInit, OnDestroy {
+      checkpoints[i].icon = this.helperService.getImage(
+        checkpoints[i].type,
+        checkpoints[i].encoding,
+        checkpoints[i].level,
+      );
+      if (i - 1 >= 0 && checkpoints[i].type > 1) {
+        if (!checkpoints[i - 1].checkpoints) {
+          checkpoints[i - 1].checkpoints = [];
+        }
+        checkpoints[i - 1].checkpoints!.push(checkpoints[i]);
+        checkpoints.splice(i, 1);
+        i--;
+      }
     }
+    report.checkpoints = checkpoints;
+    return report;
   }
 
-  findParentNode(item: any): any {
-    if (item.prevItem != undefined && item.parentElement) {
-      return this.findParentNode(item.prevItem);
-    }
+  selectReport(value: FileTreeItem): void {
+    this.selectReportEvent.emit(value.originalValue);
+  }
 
-    return item;
+  removeReport(report: any): void {
+    this.tree.removeItem(report.path);
   }
 
   closeEntireTree(): void {
     this.closeEntireTreeEvent.emit();
-    this.treeReference.clear();
+    this.tree.clearItems();
   }
 
   expandAll(): void {
-    this.treeReference.expandAll();
+    this.tree.expandAll();
   }
 
   collapseAll(): void {
-    this.treeReference.collapseAll();
+    this.tree.collapseAll();
   }
 
   downloadReports(exportBinary: boolean, exportXML: boolean): void {
     let queryString = '';
-    this.getTreeReports().forEach((report) => {
-      queryString += 'id=' + report.value.storageId + '&';
-    });
+    for (let treeReport of this.getTreeReports()) {
+      queryString += 'id=' + treeReport.storageId + '&';
+    }
     this.helperService.download(queryString, this.currentView.storageName, exportBinary, exportXML);
   }
 
   changeSearchTerm(event: KeyboardEvent): void {
-    const term: string = (event.target as HTMLInputElement).value.toLowerCase();
-    this.treeReference.getItems().forEach((item: jqwidgets.TreeItem) => {
-      const report = item.value as unknown as Report & {
-        message?: string | null;
-      };
-      if (term !== '' && report) {
-        const matching =
-          item.label?.toLowerCase() === term ||
-          report.xml?.toLowerCase().includes(term) ||
-          report.message?.toLowerCase().includes(term);
-        item.element.style.color = matching ? 'blue' : 'black';
-        item.element.style.fontWeight = matching ? 'bold' : 'normal';
-      } else {
-        item.element.style.color = 'black';
-        item.element.style.fontWeight = 'normal';
-      }
-    });
+    const term: string = (event.target as HTMLInputElement).value;
+    this.tree.searchTree(term);
   }
 
-  getTreeReports(): any[] {
-    let reports: any[] = [];
-    this.treeReference.getItems().forEach((item: any) => {
-      if (item.value?.storageId != undefined) {
-        reports.push(item);
-      }
-    });
-
-    return reports;
+  conditionalOpenFunction(item: CreateTreeItem): boolean {
+    const type = item['type'];
+    return type === undefined || type === 1 || type === 2;
```

Comment:
It would be really cool if you'd create an enum [(but not really an enum)](https://medium.com/@alex.roh/typescript-enums-what-they-are-and-why-you-should-avoid-them-20aabfcd601f) for this.

Reply:
This will be done in a later refactor step, as it will require a lot more to be refactored

Reply:
Is this noted somewhere?

Reply:
https://github.com/wearefrank/ladybug-frontend/issues/385#issue-2256457086